lib/formatters/sarif.js

```diff
@@ -0,0 +1,182 @@
+const fs = require('fs');
+const url = require('url');
+
+function getResultLevel(message) {
+  if (message.fatal || message.severity === 2) {
+    return 'error';
+  }
+  return 'warning';
+}
+
+/**
+ * This printer is heavily based on Microsoft's SARIF formatter for ESLint.
+ * https://github.com/microsoft/sarif-sdk/blob/main/src/ESLint.Formatter/sarif.js
+ */
+class SarifPrinter {
+  constructor(options = {}) {
+    this.options = options;
+    this.console = options.console || console;
+  }
+
+  print(results) {
+    const sarifLog = {
+      version: '2.1.0',
+      $schema: 'http://json.schemastore.org/sarif-2.1.0-rtm.5',
+      runs: [
+        {
+          tool: {
+            driver: {
+              name: 'ember-template-lint',
+              informationUri: 'https://github.com/ember-template-lint/ember-template-lint',
+              rules: [],
+              version: require('../../package.json').version,
+            },
+          },
+        },
+      ],
+    };
+    const sarifFiles = {};
+    const sarifArtifactIndices = {};
+    let nextArtifactIndex = 0;
+    const sarifRules = {};
+    const sarifRuleIndices = {};
+    let nextRuleIndex = 0;
+    const sarifResults = [];
+    const embedFileContents = process.env.SARIF_EMBER_TEMPLATE_LINT_EMBED === 'true';
```

Comment:
Hmm, this is somewhat odd. Should we have a better way to specify this?

Reply:
What do you suggest? Ultimately, having a way to not bloat the SARIF log with file contents seems good.

Reply:
Maybe have it leverage `--verbose` flag?

Reply:
I like.

Reply:
Done.